# Задание №1 - Перепись дохода

In [7]:
import pandas as pd
import numpy as np
import warnings
import matplotlib.pyplot as plt
import seaborn as sns
warnings.filterwarnings("ignore")
DATA_URL = "https://raw.githubusercontent.com/Yorko/mlcourse.ai/main/data/"
data = pd.read_csv(DATA_URL + "adult.data.csv")

In [8]:
data.head() 

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

# Вопросы

### 1. Сколько мужчин и женщин ( половой признак) представлены в этом наборе данных?
#### Решение:

In [11]:
data['sex'].value_counts()

sex
Male      21790
Female    10771
Name: count, dtype: int64

#### Ответ: 21790 мужчин и 10771 женщин

### 2. Каков средний возраст ( возрастная особенность) женщин?
#### Решение:

In [14]:
data[data['sex'] == 'Female']['age'].mean()

36.85823043357163

#### Ответ: 36.86 лет

### 3. Каков процент граждан Германии ( признак родной страны )?
#### Решение:

In [17]:
all_people = data.shape[0]
german = data[data['native-country'] == 'Germany'].shape[0]
print ( (german/all_people) * 100)

0.42074874850281013


#### Ответ: 0.42 %

### 4-5. Каковы среднее значение и стандартное отклонение возраста для тех, кто зарабатывает более 50 тысяч в год ( функция зарплаты ) и тех, кто зарабатывает менее 50 тысяч в год?
#### Решение:

In [20]:
data[data['salary'] == '<=50K']['age'].describe()

count    24720.000000
mean        36.783738
std         14.020088
min         17.000000
25%         25.000000
50%         34.000000
75%         46.000000
max         90.000000
Name: age, dtype: float64

In [21]:
data[data['salary'] == '>50K']['age'].describe()

count    7841.000000
mean       44.249841
std        10.519028
min        19.000000
25%        36.000000
50%        44.000000
75%        51.000000
max        90.000000
Name: age, dtype: float64

#### Ответ: для тех, кто зарабатывает менее 50 тысяч в год : 37 +- 14, ля тех, кто зарабатывает более 50 тысяч в год 44 +- 10.5

### 6. Правда ли, что люди, зарабатывающие более 50 тысяч, имеют как минимум среднее образование? ( образование – Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters or Doctorate)
#### Решение:

In [24]:
data[data['salary'] == '>50K']['education'].value_counts()

education
Bachelors       2221
HS-grad         1675
Some-college    1387
Masters          959
Prof-school      423
Assoc-voc        361
Doctorate        306
Assoc-acdm       265
10th              62
11th              60
7th-8th           40
12th              33
9th               27
5th-6th           16
1st-4th            6
Name: count, dtype: int64

#### Ответ: нет, это не правда.

### 7. Отображение статистики возраста для каждой расы  и каждого пола. Используйте groupby() и describe() . Найдите максимальный возраст мужчин данной (Amer-Indian-Eskimo) расы.
#### Решение:

In [27]:
data.groupby('race')['age'].describe()

count       mean        std   min   25%   50%   75%  \
race                                                                        
Amer-Indian-Eskimo    311.0  37.173633  12.447130  17.0  28.0  35.0  45.5   
Asian-Pac-Islander   1039.0  37.746872  12.825133  17.0  28.0  36.0  45.0   
Black                3124.0  37.767926  12.759290  17.0  28.0  36.0  46.0   
Other                 271.0  33.457565  11.538865  17.0  25.0  31.0  41.0   
White               27816.0  38.769881  13.782306  17.0  28.0  37.0  48.0   

                     max  
race                      
Amer-Indian-Eskimo  82.0  
Asian-Pac-Islander  90.0  
Black               90.0  
Other               77.0  
White               90.0

#### Ответ: 82 года.

### 8. Среди кого доля тех, кто зарабатывает много ( >50K) больше: женатых или холостых мужчин ( признак семейного положения )? Женатыми считаются те, у кого семейное положение начинается с «Married», остальные считаются холостяками.
#### Решение:

In [30]:
def Married (status):
    return 1 if 'Married' in status else 0

In [31]:
data['marital-status'].apply(Married)
data['Married'] =  data['marital-status'].apply(Married)
data[data['salary'] == '>50K']['Married'].value_counts()

Married
1    6736
0    1105
Name: count, dtype: int64

In [32]:
del data['Married']
data.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

#### Ответ: среди женатых мужчин доля тех, кто зарабатывает много ( >50K) больше.

### 9. Какое максимальное количество часов человек работает в неделю? Сколько человек работает такое количество часов и каков среди них процент тех, кто много зарабатывает (>50K)?
#### Решение:

In [35]:
max_hours = data['hours-per-week'].max()
max_hours

99

In [36]:
amount = len(data[data['hours-per-week'] == data['hours-per-week'].max()]['salary'])
amount

85

In [37]:
answ = (len(data[data['hours-per-week'] == data['hours-per-week'].max()][data['salary'] == '>50K']) / amount) * 100
answ

29.411764705882355

#### Ответ: максимальное количество часов равно 99, столько работают 85 человек, из них богаты 29%

### 10. Посчитайте среднее время работы ( часов в неделю ) для тех, кто зарабатывает мало и много ( зарплата ) по каждой стране ( родная страна ). Что это будет для Японии?
#### Решение:

In [40]:
pd.crosstab( data["native-country"],  data["salary"], values=data["hours-per-week"], aggfunc=np.mean)

salary                          <=50K       >50K
native-country                                  
?                           40.164760  45.547945
Cambodia                    41.416667  40.000000
Canada                      37.914634  45.641026
China                       37.381818  38.900000
Columbia                    38.684211  50.000000
Cuba                        37.985714  42.440000
Dominican-Republic          42.338235  47.000000
Ecuador                     38.041667  48.750000
El-Salvador                 36.030928  45.000000
England                     40.483333  44.533333
France                      41.058824  50.750000
Germany                     39.139785  44.977273
Greece                      41.809524  50.625000
Guatemala                   39.360656  36.666667
Haiti                       36.325000  42.750000
Holand-Netherlands          40.000000        NaN
Honduras                    34.333333  60.000000
Hong                        39.142857  45.000000
Hungary                     31.300000  50.000000
India                       38.233333  46.475000
Iran                        41.440000  47.500000
Ireland                     40.947368  48.000000
Italy                       39.625000  45.400000
Jamaica                     38.239437  41.100000
Japan                       41.000000  47.958333
Laos                        40.375000  40.000000
Mexico                      40.003279  46.575758
Nicaragua                   36.093750  37.500000
Outlying-US(Guam-USVI-etc)  41.857143        NaN
Peru                        35.068966  40.000000
Philippines                 38.065693  43.032787
Poland                      38.166667  39.000000
Portugal                    41.939394  41.500000
Puerto-Rico                 38.470588  39.416667
Scotland                    39.444444  46.666667
South                       40.156250  51.437500
Taiwan                      33.774194  46.800000
Thailand                    42.866667  58.333333
Trinadad&Tobago             37.058824  40.000000
United-States               38.799127  45.505369
Vietnam                     37.193548  39.200000
Yugoslavia                  41.600000  49.500000

#### Ответ: 41 и 48 часов соответственно.